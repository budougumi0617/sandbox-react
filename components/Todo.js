// @flow

import * as React from 'react';

type Props = {
  onClick: func,
  completed: boolean,
  text: string
};

type State = {};

export default class Todo extends React.Component<Props, State> {
  render() {
    return (
  <li
    onClick={this.props.onClick}
    style={ {
      textDecoration: this.props.completed ? 'line-through' : 'none'
    }}
  >
    {this.props.text}
  </li>
)};
};
