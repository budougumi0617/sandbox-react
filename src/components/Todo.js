// @flow

import * as React from 'react';

type Props = {
  onClick: (item: {}, e: Event) => void,
  completed: boolean,
  text: string
};

type State = {};

export default class Todo extends React.Component<Props, State> {
  render() {
    return (
      <li
        onClick={this.props.onClick}
        style={{
          textDecoration: this.props.completed ? 'line-through' : 'none'
        }}
      >
        {this.props.text}
      </li>
    );
  }
}
