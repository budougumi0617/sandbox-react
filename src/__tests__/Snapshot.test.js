import React from 'react';
import Todo from '../components/Todo';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Todo text="Snapshot" completed="false" />).toJSON();
  expect(tree).toMatchSnapshot();
});
