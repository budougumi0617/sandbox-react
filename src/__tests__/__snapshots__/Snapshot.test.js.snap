// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  style={
    Object {
      "textDecoration": "line-through",
    }
  }
>
  Snapshot
</li>
`;
